### PR TITLE
55-史红波-第三次作业-第一次提交

### PR DESCRIPTION
**Pull request by hongboshi1234**

### CODE REVIEW

**Comment by hongboshi1234** [comment]
55-史红波-第三次作业—第二次提交

File: Lesson-3/assignment/yours.sol [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        employees[employeeId].lastPayday = now;
+        _partialPaid(employee);
+        totalSalary += (salary - employee.salary) * 1 ether;
+        employees[employeeId].salary = salary * 1 ether;
+
+        
+    }
+    
+    function changePaymentAddress(address newEmployeeId) {
+        var employee = employees[msg.sender];
```
Comment (zchcai) [inline Lesson-3/assignment/yours.sol:167]: Better to check: (1) msg.sender is an employee (2) newEmployeeId is not an existing employee

File: Lesson-3/assignment/yours.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+#第三题
+- 第三题（加分题）：自学C3 Linearization, 求以下 contract Z 的继承线
+- contract O
+- contract A is O
+- contract B is O
+- contract C is O
+- contract K1 is A, B
+- contract K2 is A, C
+- contract Z is K1, K2
```
Comment (zchcai) [inline Lesson-3/assignment/yours.sol:204]: L(O) = [O]
L(A) = [A, O]
L(B) = [B, O]
L(C) = [C, O]
L(K2) = [K2] + merge(L(C), L(A), [C, A])
= [K2] + merge([C, O], [A, O], [C, A])
= [K2, C] + merge([O], [A, O], [A])
= [K2, C, A, O]
L(K1) = [K1] + merge(L(B), L(A), [B, A])
= [K1] + merge([B, O], [A, O], [B, A])
= [K1, B] + merge([O], [A, O], [A])
= [K1, B, A, O]

L(Z) = [Z] + merge(L(K2), L(K1), [K2, K1])
= [Z] + merge([K2, C, A, O], [K1, B, A, O], [K2, K1])
= [Z, K2] + merge([C, A, O], [K1, B, A, O], [K1])
= [Z, K2, C] + merge([A, O], [K1, B, A, O], [K1])
= [Z, K2, C, K1] + merge([A, O], [B, A, O])
= [Z, K2, C, K1, B] + merge([A, O], [A, O])
= [Z, K2, C, K1, B, A, O]

**Review by zchcai — APPROVED** [review]
Nice

**Pull request merged**